matrix-sdk/src/main/java/org/matrix/androidsdk/rest/client/MediaScanRestClient.java

```diff
@@ -167,7 +170,35 @@ public void onSuccess(String serverPublicKey) {
                 }
 
                 if (request != null) {
-                    request.enqueue(new DefaultRetrofit2CallbackWrapper<>(callback));
+                    request.enqueue(new DefaultRetrofit2CallbackWrapper<>(new SimpleApiCallback<MediaScanResult>(callback) {
+                        @Override
+                        public void onSuccess(MediaScanResult scanResult) {
+                            callback.onSuccess(scanResult);
+                        }
+
+                        @Override
+                        public void onNetworkError(Exception exception) {
+                            // Check whether the provided encrypted_body could not be decrypted.
+                            if (exception != null && exception instanceof HttpException) {
+                                HttpError error = ((HttpException) exception).getHttpError();
+                                if (error.getHttpCode() == 403) {
+                                    MediaScanError mcsError;
+                                    try {
+                                        String bodyAsString = error.getErrorBody();
+                                        mcsError = JsonUtils.getGson(false).fromJson(bodyAsString, MediaScanError.class);
+                                    } catch (Exception e) {
+                                        mcsError = null;
+                                    }
+                                    if (mcsError != null && mcsError.reason == MediaScanError.MCS_BAD_DECRYPTION) {
+                                        // The client should request again the public key of the server.
+                                        resetServerPublicKey();
+                                    }
+                                }
+                            }
+
+                            super.onNetworkError(exception);
```

Comment:
Suggestion: In the case we intercept a 403, maybe we should not call `onNetworkError` but convert it to a MatrixError (or UnexpecterError?) containing the MediaScanError to facilitate the management of this particular error for the client of the SDK

Reply:
I agree, but I'm not sure to know how to do it properly...
About this point, I observed something strange in "getServerPublicKey" (https://github.com/matrix-org/matrix-android-sdk/blob/521dd15efbbdf95e1ab70ed5a33e95656b2fa0ad/matrix-sdk/src/main/java/org/matrix/androidsdk/rest/client/MediaScanRestClient.java#L97)
Are we sure that "onMatrixError" is called on 404 error?

Reply:
Should be done thanks to this: https://github.com/matrix-org/matrix-android-sdk/blob/develop/matrix-sdk/src/main/java/org/matrix/androidsdk/rest/callback/RestAdapterCallback.java#L226

Reply:
But I was not able to test it

Reply:
ok